mycore-neo4j/src/main/java/org/mycore/mcr/neo4j/utils/MCRNeo4JDatabaseDriver.java

```diff
@@ -0,0 +1,381 @@
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.mycore.common.config.MCRConfiguration2;
+import org.mycore.mcr.neo4j.datamodel.metadata.neo4jtojson.Neo4JMetaData;
+import org.mycore.mcr.neo4j.datamodel.metadata.neo4jtojson.Neo4JNodeJsonRecord;
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.GraphDatabase;
+import org.neo4j.driver.Query;
+import org.neo4j.driver.Record;
+import org.neo4j.driver.Result;
+import org.neo4j.driver.Session;
+import org.neo4j.driver.Value;
+import org.neo4j.driver.types.Node;
+import org.neo4j.driver.types.Path;
+import org.neo4j.driver.types.Relationship;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The MCRNeo4JDatabaseDriver class is a Java driver implementation for connecting to a Neo4j database. It provides
+ * methods to establish a connection, test connection settings, and execute queries.
+ *
+ * @author Andreas Kluge
+ */
+public class MCRNeo4JDatabaseDriver {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    private static final String LANGUAGE = MCRConfiguration2.getString("MCR.Metadata.DefaultLang").orElse("de");
+
+    private static MCRNeo4JDatabaseDriver instance = null;
+
+    private final String url;
+
+    private String user;
+
+    private String password;
+
+    private Driver driver;
+
+    /**
+    * Constructs an MCRNeo4JDatabaseDriver object with connection settings loaded from a configuration file.
+    */
+    public MCRNeo4JDatabaseDriver() {
+        this.url = MCRConfiguration2.getString(NEO4J_CONFIG_PREFIX + "ServerURL").orElse("");
+        this.user = MCRConfiguration2.getString(NEO4J_CONFIG_PREFIX + "user").orElse("");
+        this.password = MCRConfiguration2.getString(NEO4J_CONFIG_PREFIX + "password").orElse("");
+    }
+
+    /**
+    * Constructs an MCRNeo4JDatabaseDriver object with the specified URL.
+    *
+    * @param url the URL of the Neo4j database
+    */
+    public MCRNeo4JDatabaseDriver(String url) {
+        this.url = url;
+    }
+
+    /**
+    * Constructs an MCRNeo4JDatabaseDriver object with the specified URL, username, and password. It also creates a
+    * connection to the database.
+    *
+    * @param url      the URL of the Neo4j database
+    * @param user     the username for the database connection
+    * @param password the password for the database connection
+    */
+    public MCRNeo4JDatabaseDriver(String url, String user, String password) {
+        this.url = url;
+        this.user = user;
+        this.password = password;
+        this.createConnection();
```

Comment:
nur in diesem Konstruktor wird `createConnection()` aufgerufen

Reply:
der Constructor wurde entfernt